### PR TITLE
Fix: Misc fixes for verl 0.7.1

### PR DESCRIPTION
**Pull request by kylemontgomery1**
## Summary

Misc bug fixes for verl 0.7.1 integration

## Type of change

- [ ] Feature
- [x] Fix
- [ ] Docs
- [ ] Refactor
- [ ] Example / Project
- [ ] Infra / CI

## What changed

1. support both old and new worker implementations in agent workflow engine
2. add super().__init__() to experimental verl engine
3. fix verl backend (unified trainer) to pass validation vlag to engine
4. adopt verl deps from branch verl_071_deps
5. add countdown example for sync verl with unified trainer

## Validation
https://wandb.ai/agentica/rllm-countdown
Three runs all track the same val performance:
1. agent workflow train + legacy verl worker
2. agent workflow train + new verl worker
3. unified trainer sync verl (only support new verl worker)

- [x] `pre-commit run --all-files`
- [ ] Targeted tests: `pytest ...`
- [x] Manual validation performed
- [ ] Not run (reason below)

Validation details:
- passes pre-commit tests

## Breaking changes / migration notes

None

- None

## Docs / examples

- [x] Not needed
- [ ] Updated docs
- [x] Updated examples
- [ ] Follow-up docs needed

## Related issues / PRs

- Fixes #
- Related to #
- Stacked on / depends on #

### CODE REVIEW

**Comment by kylemontgomery1** [comment]
@listar2000 @luyuzhe111

**Comment by listar2000** [comment]
LGTM! I think this is mergeable since the logic is pretty clear that we are just supporting two modes for legacy workers in Verl. Also the VerlBackend is targeting at deprecating the legacy workers so is not affected.

File: rllm/trainer/verl/agent_workflow_trainer.py [hunk 1]
```diff
@@ -457,26 +468,31 @@ def fit_agent(self):
 
                     # implement critic warmup
                     if self.config.trainer.critic_warmup <= self.global_steps:
-                        # verl 0.7.1 new worker path: update_actor needs training metadata
-                        ppo_mini_batch_size = self.config.actor_rollout_ref.actor.ppo_mini_batch_size * self.config.actor_rollout_ref.rollout.n
+                        # Mirror verl's RayPPOTrainer._update_actor: meta_info goes onto the
+                        # DataProto for the legacy worker; the new EngineWorker path needs a
+                        # TensorDict in no-padding format with training metadata set via
+                        # tu.assign_non_tensor.
+                        batch.meta_info["multi_turn"] = self.config.actor_rollout_ref.rollout.multi_turn.enable
```
Comment (luyuzhe111) [inline rllm/trainer/verl/agent_workflow_trainer.py:475]: wondering what the impact of this meta info arg is?

Comment (kylemontgomery1) [reply rllm/trainer/verl/agent_workflow_trainer.py:475]: I don't think "multi_turn" is actually used by verl, I added to just to be consistent with verl's trainer. The "temperature" arg is important, not setting it would result in training against unscaled logits (if your training temperature is not 1.0).
